flink-runtime/src/main/java/org/apache/flink/runtime/state/StateSnapshotTransformer.java

```diff
@@ -120,6 +122,17 @@ public ListStateSnapshotTransformer(StateSnapshotTransformer<T> entryValueTransf
 		}
 	}
 
+	class ListStateSnapshotTransformFactory<T> extends StateSnapshotTransformFactoryWrapAdaptor<T, List<T>> {
```

Comment:
Why do all these factory classes live within the `StateSnapshotTransformer`?

Reply:
True, I can leave only interfaces here and move all facility classes into a `StateSnapshotTransformers` class.